src/main/kotlin/com/sourcegraph/cody/chat/ui/ContextFilesPanel.kt

```diff
@@ -54,48 +60,85 @@ class ContextFilesPanel(
       val accordionSection = AccordionSection("Read ${filesAvailableInEditor.size} files")
       accordionSection.isOpaque = false
       accordionSection.border = EmptyBorder(margin)
-      filesAvailableInEditor.forEachIndexed { index, file: VirtualFile ->
-        val filePanel = createFileWithLinkPanel(file)
+      filesAvailableInEditor.forEachIndexed { index, file: Pair<VirtualFile, ContextFileFile?> ->
+        val filePanel = createFileWithLinkPanel(file.first, file.second)
         accordionSection.contentPanel.add(filePanel, index)
       }
       add(accordionSection, BorderLayout.CENTER)
     }
   }
 
   @RequiresEdt
-  private fun createFileWithLinkPanel(file: VirtualFile): JPanel {
+  private fun createFileWithLinkPanel(
+      file: VirtualFile,
+      contextFileFile: ContextFileFile?
+  ): JPanel {
     val projectRelativeFilePath = file.path.removePrefix(project.basePath ?: "")
     val anAction =
         object : DumbAwareAction() {
           override fun actionPerformed(anActionEvent: AnActionEvent) {
-            logger.info(
-                "Opening a file from the used context (projectRelativeFilePath=$projectRelativeFilePath, file=$file)")
-            FileEditorManager.getInstance(project).openFile(file, /*focusEditor=*/ true)
+            val logicalLine = contextFileFile?.range?.start?.line ?: 0
+            OpenFileDescriptor(project, file, logicalLine, /* logicalColumn= */ 0)
+                .navigate(/* requestFocus= */ true)
           }
         }
-    val goToFile = AnActionLink("@$projectRelativeFilePath", anAction)
+    val actionText = getActionTextForFileLinkAction(contextFileFile, projectRelativeFilePath)
+    val goToFile = AnActionLink(actionText, anAction)
+    goToFile.toolTipText = actionText
     val panel = JPanel(BorderLayout())
     panel.isOpaque = false
     panel.border = JBUI.Borders.emptyLeft(3)
     panel.add(goToFile, BorderLayout.PAGE_START)
     return panel
   }
 
+  private fun getActionTextForFileLinkAction(
+      contextFileFile: ContextFileFile?,
+      projectRelativeFilePath: String
+  ): String {
+    val (startLine, endLine) =
+        contextFileFile?.range?.let {
+          // We need to .plus(1) since the ranges use 0-based indexing
+          // but IntelliJ presents it as 1-based indexing.
+          Pair(it.start.line.plus(1), it.end.line.plus(1))
```

Comment:
Cannot you use `Range` instead of pair?
You could even add method like `intellijRange` on the `Range` class so you won't need to do that mapping there, and just call  `intellijRange` where you need it.